src/asciidoctorFindIncludeProcessor.ts

```diff
@@ -0,0 +1,49 @@
+interface IncludeEntry {
+  name: string
+  position: number
+  length: string
+}
+
+interface IncludeItem {
+  [index: number]: IncludeEntry
+}
+
+interface IncludeItems extends Array<IncludeItem>{}
+
+let baseDocIncludes: IncludeItems[] = []
+
+function findIncludeProcessor () {
+  const self = this
+
+  self.handles(function (_target) {
+    return true
+  })
+
+  self.process(function (doc, reader, target, attrs) {
+    // We don't meaningfully process the includes, we just want to identify
+    // their line number and path if they belong in the base document
+    if (reader.path === '<stdin>') {
+      baseDocIncludes.push([target, reader.lineno - 1, target.length])
+    }
+    return reader.pushInclude(['nothing'], target, target, 1, attrs)
+  })
```

Comment:
I don't get why do we need to do that? Using a regular expression on the text input seems enough no?

Reply:
No, the input text might for instance use attribute substitutions.